delphi/translators/for2py/genModFileLog.py

```diff
@@ -152,14 +159,211 @@ def populate_mappers(file_path, file_to_mod_mapper, mod_to_file_mapper,
+                                    variables  = variables.partition('=')[0].strip()
+
+                            var_list = variables.split(',')
+                            for var in var_list:
+                                # Search for an implicit array variable declaration
+                                arrayVar = re.match("(?P<var>\w*)\s*\(", var)
+                                if arrayVar:
+                                    var = arrayVar['var']
+                                    var_type = "Array"
+                                # Map each subroutine argument with its type
+                                if (
+                                        current_subr in module_summary[current_modu]
+                                        and var.strip() in module_summary[current_modu][current_subr]
+                                ):
+                                    module_summary[current_modu][current_subr][var.strip()] = var_type
+                    else:
+                        pass
+
+                    if interface:
+                        current_intr  = interface['name'].strip()
+                    elif end_intr or endintr:
+                        current_intr = None
+                        if isProcedure:
+                            isProceddure = False
+                    elif current_intr:
+                        if "procedure" in line:
+                            first_function = line.strip().split(' ')[-1].replace(',','')
+                            procedure_functions[current_modu][current_intr] = {first_function:None}
+                            isProcedure = True
+                        elif isProcedure:
+                            pFunc = line.strip().split(' ')[-1]
+                            if pFunc:
+                                pFunc = pFunc.replace(',','')
+                                procedure_functions[current_modu][current_intr][pFunc] = None
+                    else:
+                        pass
+
+                    if func:
+                        func_name = func["name"]
+                        func_args = func["args"]
+                    else:
+                        pass
+
+                    if end_dtype or enddtype:
+                        if current_modu not in derived_types:
+                            if end_dtype:
+                                derived_types[current_modu] = [end_dtype['name'].strip()]
+                            elif enddtype:
+                                derived_types[current_modu] = [enddtype['name'].strip()]
+                        else:
+                            if end_dtype:
+                                derived_types[current_modu].append(end_dtype['name'].strip())
+                            elif enddtype:
+                                derived_types[current_modu].append(enddtype['name'].strip())
+
+                line  = f.readline().lower()
```

Comment:
I'm not a fan of a giant wall of code.  The code for processing a file to compute information about modules should be in a separate function.

Reply:
@skdebray I've separated out the function into multiple based on purposes and scope, such as extracting information from a subroutine, module, and derived types. If you think I should segment them further, please let me know.